R/exclude.R

```diff
@@ -98,26 +178,69 @@ normalize_exclusions <- function(x, normalize_path = TRUE,
 
   # no named parameters at all
   if (is.null(names(x))) {
-    x <- structure(relist(rep(Inf, length(x)), x), names = x)
+    bad <- vapply(
+      seq_along(x),
+      function(i) {
+        (!is.character(x[[i]]) || length(x[[i]]) != 1L)
+      },
+      logical(1L)
+    )
+
+    if (any(bad)) {
+      stop("Full file exclusions must be character vectors of length 1. items: ",
+           toString(which(bad)),
+           " are not!",
+           call. = FALSE)
+    }
+    # x is a character vector (or list) of file names
+    # Normalize to list(<filename> = list(Inf), ...)
+    x <- structure(rep(list(Inf), length(x)), names = x)
   } else {
     unnamed <- names(x) == ""
     if (any(unnamed)) {
 
       # must be character vectors of length 1
-      bad <- vapply(seq_along(x),
+      bad <- vapply(
+        seq_along(x),
         function(i) {
-          unnamed[i] & (!is.character(x[[i]]) | length(x[[i]]) != 1)
+          unnamed[i] & (!is.character(x[[i]]) | length(x[[i]]) != 1L)
         },
-        logical(1))
+        logical(1L)
+      )
 
       if (any(bad)) {
         stop("Full file exclusions must be character vectors of length 1. items: ",
-             paste(collapse = ", ", which(bad)),
+             toString(which(bad)),
              " are not!",
              call. = FALSE)
       }
       names(x)[unnamed] <- x[unnamed]
-      x[unnamed] <- Inf
+      x[unnamed] <- rep(list(list(Inf)), sum(unnamed))
```

Comment:
any reason to use `rep_len` below but not here?

Reply:
Just an oversight.